jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -955,10 +937,6 @@ private Object call(Class<?> oClass, String methodName, Object obj, Args args) t
                 Object[] arguments = args.applyTo(method);
                 if (arguments == null)
                     continue;
-                if (Modifier.isStatic(method.getModifiers()) != (obj == null))
-                    continue;
-                if ((obj == null) && method.getDeclaringClass() != oClass)
-                    continue;
```

Comment:
Is this what was preventing `testVirtualCallWithClassAttribute` from working?

Reply:
Not only. There were places where if the `class` attribute was present, then `obj` was nulled out, which also I removed.